### PR TITLE
Sketcher: clean-up geometry history configuation

### PR DESCRIPTION
**Pull request by pinkavaj**
Keep code almost 1:1 with LS3 branch, just ensure the variable is initialized and code is documented

Follow-up for #16337

### CODE REVIEW

**Review by bgbsww — COMMENTED** [review]
Overall, this does no harm and looks correct.

However, I note searching the LS3 source code that GeometryHistoryLevel appears to never be set anywhere, and not appear in any dialogs, so this appears to be an unused option that should either be completely removed, or surfaced in a UI.
It might have been placed historically and then determined not to be needed.

The prior fix was required to stop the uninitialized variable bug, but I think that this one should either make this settable because we understand that it does something useful or remove it altogether.

If it does do something useful, then maybe a test demonstrating that is warranted?

My gut says this was useful during TNP mitigation development, but ultimately was not needed.

File: src/Mod/Sketcher/App/SketchObject.cpp [hunk 1]
```diff
@@ -212,6 +212,7 @@ void SketchObject::setupObject()
 {
     ParameterGrp::handle hGrpp = App::GetApplication().GetParameterGroupByPath(
             "User parameter:BaseApp/Preferences/Mod/Sketcher");
+    geoHistoryLevel = hGrpp->GetInt("GeometryHistoryLevel",1);
```
[the remark this answers is not in the document]
Comment (pinkavaj) [reply src/Mod/Sketcher/App/SketchObject.cpp:215]: Sure, some adjustmenst will be needed after rebase ...

I already have enabled pre-commit, lets continue the discussion once rebase ...

**Comment by luzpaz** [comment]
> However, I note searching the LS3 source code that GeometryHistoryLevel appears to never be set anywhere, and not appear in any dialogs, so this appears to be an unused option that should either be completely removed, or surfaced in a UI.
It might have been placed historically and then determined not to be needed.

@realthunder can you weigh in ?

File: src/Mod/Sketcher/App/SketchObject.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinkavaj) [reply src/Mod/Sketcher/App/SketchObject.cpp]: @bgbsww It looks like the path for this file is now disabled in `.pre-commit-config.yaml` ... :shrug: 

@bgbsww It also looks like the initialisation cannot be removed as you suggested. The `getHistoryLevel` is set in `setupObject()` but it looks like this function is no called (or is called too late) when objects are created during loading document from file. I'm not sure if the `setupObject()` gets called during load from file ... investigating, any idea? Should it be called?

**Comment by maxwxyz** [comment]
any news on this PR, what is missing?

**Comment by CalligaroV** [comment]
@maxwxyz, @pinkavaj, I see this PR now for the first time.

ATM I can't share any comments, I'll need some time to study it (or better, its effect as the modified code isn't much).

Unfortunately my task list is pretty long with issues that for me are quite complex to analyze, therefor I don't know when I'll be able to look at this and give any useful feedback.

As for other issues/PRs, if I don't give updates don't wait for me to merge but don't hesitate to ping me again.
Maybe next time I'll have more time to study it.

TIA!

**Comment by pinkavaj** [comment]
I did some testing and playing around. My opinion is not to make it configurable and keep only the code path with the  functionality enabled. IMHO it is not much benefit to add such low-level configuration options, the complexity is not worth the benefits.

**Comment by CalligaroV** [comment]
@maxwxyz, @pinkavaj I gave a more careful look at this PR and related comments.

@pinkavaj, I agree with your https://github.com/FreeCAD/FreeCAD/pull/16520#issuecomment-2465806764: ATM making `geoHistoryLevel` via Parameter doesn't look worth the effort and probably not very useful.

The code affected by that variable and executed by now both on main and on LS3 (with the default value, that is **1**) is exactly the same.

Therefor IMO if we want we can move on with this PR keeping only the non-Parameter code.
I.e. only the initialization of `geoLastId` in the constructor to make the linter happy.

Otherwise we can simply close this PR, leaving untouched non-aligned code in main aligned to LS3 (which I personally would prefer as in general this makes easier comparing the 2 codebases)

Any case for me both ways are ok, the only difference is for my personal opinion, therefor I leave up to you to decide how to proceed with this one.

Thanks!

**Comment by pinkavaj** [comment]
> @maxwxyz, @pinkavaj I gave a more careful look at this PR and related comments.
> 
> @pinkavaj, I agree with your [#16520 (comment)](https://github.com/FreeCAD/FreeCAD/pull/16520#issuecomment-2465806764): ATM making `geoHistoryLevel` via Parameter doesn't look worth the effort and probably not very useful.
> 
> The code affected by that variable and executed by now both on main and on LS3 (with the default value, that is **1**) is exactly the same.
> 
> Therefor IMO if we want we can move on with this PR keeping only the non-Parameter code. I.e. only the initialization of `geoLastId` in the constructor to make the linter happy.
> 
> Otherwise we can simply close this PR, leaving the untouched non-aligned code in main equivalent to LS3's (which I personally would prefer as in general this makes easier comparing the 2 codebases)
> 
> Any case for me both ways are ok, the only difference is for my personal opinion, therefor I leave up to you to decide how to proceed with this one.
> 
> Thanks!

Would make sense to keep the code aligned with LS3, except to add oneline comment to make some info why the code is here in the state in which it is, for future developers who might stumble upon this (not like there will be many of them, but still this is not one-man show ...)

**Comment by maxwxyz** [comment]
@chennes what is your opinion on this PR?

File: src/Mod/Sketcher/App/SketchObject.h [hunk 2]
```diff
@@ -1062,7 +1062,8 @@ class SketcherExport SketchObject: public Part::Part2DObject
     // mapping from Geometry[*].Id to index of Geometry
     std::map<long, int> geoMap;
 
-    int geoHistoryLevel;
+    // keep geoHistoryLevel and the code who ise it for easier porting of stuff from LS3 branch
```
Comment (CalligaroV) [inline src/Mod/Sketcher/App/SketchObject.h:1065]:
```suggestion
    // keep geoHistoryLevel and the code who use it for easier porting of stuff from LS3 branch
```

**Review by CalligaroV — APPROVED** [review]

**Pull request merged**